app/models/subnet.rb

```diff
@@ -23,7 +23,9 @@ class Subnet < ActiveRecord::Base
 
   # This casts Subnet to Subnet::Ipv4 if no type is set
   def self.new(*attributes, &block)
-    return Subnet::Ipv4.new_without_cast(*attributes, &block) if self == Subnet
+    type = nil
```

Comment:
Nope, `type` is a db field of this model so it's nil by default. Try removing line 26, then putting a debugger breakpoint and you'll see `type` is nil.